public/scripts/menu.js

```diff
@@ -0,0 +1,86 @@
+miniDom = {};
+
+miniDom.hasClass = function(el, className){
+    if (el.classList)
+        return el.classList.contains(className);
+    else
+        return new RegExp('(^| )' + className + '( |$)', 'gi').test(el.className);
+}
+
+miniDom.removeClass = function(el, className){
+    if (el.classList)
+      el.classList.remove(className);
+    else
+      el.className = el.className.replace(new RegExp('(^|\\b)' + className.split(' ').join('|') + '(\\b|$)', 'gi'), ' ');
+}
+
+miniDom.matches = function(el, selector) {
+    // shortcut to use matches globally
+    if(typeof selector === "undefined"){
+        selector = el;
+        el = document.body;
+    }
+    return (el.matches || el.matchesSelector || el.msMatchesSelector || el.mozMatchesSelector || el.webkitMatchesSelector || el.oMatchesSelector).call(el, selector);
+};
+
+miniDom.toggleClass = function(el, className) {
+    if (el.classList) {
+      el.classList.toggle(className);
+    } else {
+      var classes = el.className.split(' ');
+      var existingIndex = classes.indexOf(className);
+
+      if (existingIndex >= 0)
+        classes.splice(existingIndex, 1);
+      else
+        classes.push(className);
+
+      el.className = classes.join(' ');
+    }
+}
+
+miniDom.parents = function (elem, selector) {
+    var parents = [];
+    while (elem.parentNode != document) {
+        if(miniDom.matches(elem.parentNode, selector))
+            parents.push(elem.parentNode);
+        elem = elem.parentNode;
+    }
+    return parents;
+};
+
+if (typeof cssmenu_no_js === 'undefined') {
+    var open_main_item = null;
+
+    var expandToggles = document.getElementsByClassName("expand-toggle");
+    [].forEach.call(expandToggles, function(expandToggle){
+        expandToggle.addEventListener("click", function(e) {
+            var container = miniDom.parents(this, '.expand-container')[0];
+            miniDom.toggleClass(container, 'open');
+
+            /* In the main menu, let only one dropdown be open at a
+            time. Also close any open main menu dropdown when clicking
+            elsewhere. */
+            if (open_main_item !== container && open_main_item !== null) {
+                miniDom.removeClass(open_main_item, "open");
+            }
+            var clicking_main_bar = miniDom.parents(container, "#top").length > 0;
+            var clicking_hamburger = this === miniDom.matches('.hamburger')[0];
+            if (clicking_main_bar && !clicking_hamburger) {
+                open_main_item = miniDom.hasClass(container, 'open')
+                    ? container : null;
+            }
+            return false;
+        });
+    });
+
+    // close window on clicks to other regions
+    window.addEventListener("click", function(e) {
+        var clicking_main_bar = miniDom.parents(e.target, "#top").length > 0;
```

Comment:
Unfortunately we can't make it that simple as 
1) the code would allow multiple menus to be open at the same time (we need `remove`)
2) the hamburger (menu button for mobile devices is a special case)
3) `return false` is not the entire game - we need to use `stopPropagation` in case we want to prevent it from bubbling up

Thanks to your amazing feedback the menu code now looks a lot cleaner :)